packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/template/ConvenienceMethodTemplateBase.java

```diff
@@ -822,9 +826,12 @@ private static boolean isMultipartModel(IType type) {
         Map<MethodParameter, MethodParameter> parameterMap = new LinkedHashMap<>();
         List<MethodParameter> convenienceParameters = getParameters(convenienceMethod, true);
         Map<String, MethodParameter> clientParameters = getParameters(protocolMethod, false).stream()
-            .collect(Collectors.toMap(MethodParameter::getSerializedName, Function.identity()));
+            .collect(Collectors.toMap(key -> key.getSerializedName() == null ? key.getName() : key.getSerializedName(),
+                Function.identity()));
         for (MethodParameter convenienceParameter : convenienceParameters) {
-            String name = convenienceParameter.getSerializedName();
+            String name = convenienceParameter.getSerializedName() == null
+                ? convenienceParameter.getName()
+                : convenienceParameter.getSerializedName();
```

Comment:
Using `name` alone could be risky, when `serializedName` is null.

Do we know what case would cause this?